### PR TITLE
DOC: Enforce Numpy Docstring Validation for pandas.Grouper

### PR DESCRIPTION
**Pull request by tuhinsharma121**
- [ ] xref #58067 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/grouper.py [hunk 1]
```diff
@@ -71,52 +71,18 @@ class Grouper:
         Currently unused, reserved for future use.
     **kwargs
         Dictionary of the keyword arguments to pass to Grouper.
-    key : str, defaults to None
```
Comment (mroeschke) [inline pandas/core/groupby/grouper.py:74]: These should remain

File: pandas/core/groupby/grouper.py [hunk 2]
```diff
@@ -71,52 +71,18 @@ class Grouper:
...
-        row/column will be dropped. If False, NA values will also be treated as
-        the key in groups.
 
     Returns
     -------
     Grouper or pandas.api.typing.TimeGrouper
         A TimeGrouper is returned if ``freq`` is not ``None``. Otherwise, a Grouper
         is returned.
 
+    See ALso
```
Comment (mroeschke) [inline pandas/core/groupby/grouper.py:81]:
```suggestion
    See Also
```

Comment (tuhinsharma121) [reply pandas/core/groupby/grouper.py:81]: fixed it now.

File: pandas/core/groupby/grouper.py (hunk 1 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/grouper.py:74]: I have added these back with an additional indentation else

`scripts/validate_docstrings.py --format=actions pandas.Grouper` is giving the following error.

```
################################################################################
################################## Validation ##################################
################################################################################

1 Errors found for `pandas.Grouper`:
        PR02    Unknown parameters {'key', 'label', 'sort', 'closed', 'convention', 'dropna', 'freq', 'level', 'offset', 'origin'}

```

With indentation its working fine

```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Grouper" correct. :)
```

Comment (tuhinsharma121) [reply pandas/core/groupby/grouper.py:74]: let me know if this is fine now.

Comment (mroeschke) [reply pandas/core/groupby/grouper.py:74]: I don't think this should be adjusted just to appease the current docstring checker. IMO it looks a bit confusing that this is nested under kwargs.

I would suggest avoid touching these parameters as it probably would be better to make these keywords permanement over `*args, **kwargs`

File: pandas/core/groupby/grouper.py [hunk 3]
```diff
@@ -71,52 +71,58 @@ class Grouper:
...
+            the key in groups.
 
     Returns
     -------
     Grouper or pandas.api.typing.TimeGrouper
         A TimeGrouper is returned if ``freq`` is not ``None``. Otherwise, a Grouper
         is returned.
 
+    See Also
+    ----------
```
Comment (mroeschke) [inline pandas/core/groupby/grouper.py:122]:
```suggestion
    See Also
    --------
```

File: pandas/core/groupby/grouper.py [hunk 4]
```diff
@@ -71,52 +71,58 @@ class Grouper:
...
+    GroupBy : For grouping and aggregating relational data.
+    GroupByPlot : Class implementing the .plot attribute for groupby objects.
```
Comment (mroeschke) [inline pandas/core/groupby/grouper.py:124]:
```suggestion
    Series.groupby : Apply a function groupby to a Series.
    DataFrame.groupby : Apply a function groupby.
```

File: pandas/core/groupby/grouper.py (hunk 1 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/grouper.py:74]: restored it to upstream main.

File: pandas/core/groupby/grouper.py (hunk 3 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/grouper.py:122]: I see. The number of - should be equal to the number of characters in the text including whitespace. Fixed it.

File: ci/code_checks.sh [hunk 5]
```diff
@@ -153,7 +153,7 @@ if [[ -z "$CHECK" || "$CHECK" == "docstrings" ]]; then
         -i "pandas.DatetimeTZDtype SA01" \
         -i "pandas.DatetimeTZDtype.tz SA01" \
         -i "pandas.DatetimeTZDtype.unit SA01" \
-        -i "pandas.Grouper PR02,SA01" \
+        -i "pandas.Grouper PR02" \
```
Comment (tuhinsharma121) [inline ci/code_checks.sh:156]: I have kept the PR02 else the CI fails.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**